src/Interpreters/Context.cpp

```diff
@@ -3777,21 +3778,28 @@ void Context::updateStorageConfiguration(const Poco::Util::AbstractConfiguration
             try
             {
                 shared->merge_tree_storage_policy_selector = shared->merge_tree_storage_policy_selector->updateFromConfig(
-                    config, "storage_configuration.policies", shared->merge_tree_disk_selector);
+                    config, "storage_configuration.policies", shared->merge_tree_disk_selector, disks_to_reinit);
             }
             catch (Exception & e)
             {
                 LOG_ERROR(
                     shared->log, "An error has occurred while reloading storage policies, storage policies were not applied: {}", e.message());
             }
         }
+
+        if (!disks_to_reinit.empty())
+        {
+            LOG_INFO(shared->log, "Initializing disks: ({}) for all tables", fmt::join(disks_to_reinit, ", "));
+            DatabaseCatalog::instance().triggerReloadDisksTask(disks_to_reinit);
```

Comment:
It may be triggered by a user query (SYSTEM RELOAD CONFIG). If the server has many tables, reloading disks may take long time. So I think it's better to be a background task.

Reply:
I've raised the question because in the test you have `sleep(1)` (likely because of this).

But the concern about lots of disks/tables make sense I guess, though, personally I, wound prefer to see any errors from the context of RELOAD query instead of looking into the logs...

Besides it should be the case only when some disks had been added, while this is not a common thing I would say.